src/pages/p/[handle].page.tsx

```diff
@@ -1,46 +1,166 @@
-import useSymbolOfERC20 from 'hooks/SymbolOfERC20'
 import useTerminalOfProject from 'hooks/v1/contractReader/TerminalOfProject'
 import useTokenAddressOfProject from 'hooks/v1/contractReader/TokenAddressOfProject'
-import useUriOfProject from 'hooks/v1/contractReader/UriOfProject'
-import { useCurrencyConverter } from 'hooks/CurrencyConverter'
+import { consolidateMetadata, ProjectMetadataV4 } from 'models/project-metadata'
 import { V1CurrencyOption } from 'models/v1/currencyOption'
-import { useMemo } from 'react'
+import { GetStaticPaths, GetStaticProps, InferGetStaticPropsType } from 'next'
 import { useRouter } from 'next/router'
-import { getTerminalName, getTerminalVersion } from 'utils/v1/terminals'
 import { V1CurrencyProvider } from 'providers/v1/V1CurrencyProvider'
+import { useMemo } from 'react'
+import { ipfsCidUrl } from 'utils/ipfs'
 import { V1CurrencyName } from 'utils/v1/currency'
-import NewDeployNotAvailable from 'components/NewDeployNotAvailable'
-import Project404 from 'components/Project404'
-import { usePageTitle } from 'hooks/PageTitle'
+import ScrollToTopButton from 'components/ScrollToTopButton'
 
-import useOwnerOfProject from 'hooks/v1/contractReader/OwnerOfProject'
+import { getTerminalName, getTerminalVersion } from 'utils/v1/terminals'
+import { paginateDepleteProjectsQueryCall } from 'utils/apollo'
+
+import Loading from '../../components/Loading'
+import V1Project from '../../components/v1/V1Project'
 
-import { layouts } from 'constants/styles/layouts'
 import { projectTypes } from 'constants/v1/projectTypes'
+import { layouts } from 'constants/styles/layouts'
 import { V1ArchivedProjectIds } from 'constants/v1/archivedProjects'
 
-import Loading from '../../components/Loading'
-import V1Project from '../../components/v1/V1Project'
+export const getStaticPaths: GetStaticPaths = async () => {
+  if (process.env.BUILD_CACHE_V1_PROJECTS === 'true') {
+    const projects = await paginateDepleteProjectsQueryCall({
+      variables: {
+        where: { cv_in: ['1', '1.1'] },
+      },
+    })
+    const paths = projects
+      .map(({ handle }) =>
+        handle
+          ? {
+              params: { handle },
+            }
+          : undefined,
+      )
+      .filter((i): i is { params: { handle: string } } => !!i)
+    return { paths, fallback: true }
+  }
+
+  return { paths: [{ params: { handle: 'juicebox' } }], fallback: true }
+}
 
-export default function V1Dashboard() {
+export const getStaticProps: GetStaticProps<{
+  metadata: ProjectMetadataV4
+  handle: string
+}> = async context => {
+  if (!context.params) throw new Error('params not supplied')
+  const handle = context.params.handle as string
+  const projects = await paginateDepleteProjectsQueryCall({
+    variables: {
+      where: { cv_in: ['1', '1.1'], handle },
+    },
+  })
+  if (!projects[0]?.metadataUri) {
+    throw new Error(`Failed to load metadata uri for ${context.params}`)
+  }
+  const url = ipfsCidUrl(projects[0].metadataUri)
```

Comment:
**suggestion** This feels like it could get inefficient.

Perhaps we could add `first: 1` to only fetch 1 result from the subgraph.

Ideally we do a contract read here though.

WDYT?

Reply:
I'll trial `first: 1` and see if it works as expected. I am less inclined to do much optimisation around v1 stuff (we do the contract read in v2)